pkg/openstack/loadbalancer.go

```diff
@@ -1150,11 +1150,15 @@ func (lbaas *LbaasV2) buildMonitorCreateOpts(svcConf *serviceConfig, port corev1
 	if port.Protocol == corev1.ProtocolUDP {
 		opts.Type = "UDP-CONNECT"
 	}
-	if svcConf.healthCheckNodePort > 0 && lbaas.canUseHTTPMonitor(port) {
-		opts.Type = "HTTP"
-		opts.URLPath = "/healthz"
-		opts.HTTPMethod = "GET"
-		opts.ExpectedCodes = "200"
+	if svcConf.healthCheckNodePort > 0 {
```

Comment:
I'd make the opts return right away, e.g. 
```suggestion
	if svcConf.healthCheckNodePort == 0 {
		return opts
	}
```

and place the rest if/then outside the `if svcConf.healthCheckNodePort > 0 {` condition.